### ISSUE TITLE
SSH fails after reload when shared adapter uses dhcp

### ISSUE THREAD

**Issue by dx9** [Parallels/vagrant-parallels#194]
When a guest uses DHCP, it is assigned a new IP on each reload. The SSH logic does not detect the IP change. SSH fails because it attempts to connect to the original IP that is no longer valid.

Would be nice if it either detected the new ip or issued the same IP across reloads.

**Comment by legal90** [comment]
@dx9 Please, show your Vagrantfile and tell what the box name&version you are using. 
Box version can be figured out by `vagrant box list`.

**Comment by dx9** [comment]
Hey @legal90, here's some info on the setup I'm using.

Built a box file for parallels using https://github.com/dx9/os-packer. Here is a link to the box https://www.dropbox.com/s/b17tj5wwr6i7s1h/rancheros_parallels.box?dl=0

To reproduce the issue tweak the Vagrantfile to load the box.
1) vagrant up ... it will take some time to download the os and ssh in on first boot
2) rancherctl os upgrade (this will download latest os with some nice ip prints on the console, so you can see the IP incrementing on eth0 with each reload)
3) do the reboot
4) vagrant ssh --debug will show the attempt to login to the old IP

Thanks for taking a look!

**Comment by Tomdarkness** [comment]
I'm having the same issue. See the attached screenshot. Vagrant is trying to connect to 10.211.55.38 but as you can see from the screenshot eth0 on in the VM has IP 10.211.55.43.

![screen shot 2015-06-26 at 12 59 18](https://cloud.githubusercontent.com/assets/1273529/8376849/6dfe4cf0-1c03-11e5-8b2a-0148ce872398.png)

**Comment by Tomdarkness** [comment]
Looking into this it appears the Parallel's DHCP server has two leases for the same mac address which is causing the plugin to use the wrong IP (looked at source - plugin gets the IP for SSH by looking in the DHCP lease file for the VM network adapter's mac).

In `/Library/Preferences/Parallels/parallels_dhcp_leases` I have:

```
...
10.211.55.38="1435319952,1800,001c4270603b,ff2fd08bae00020000ab11873bec156a374b12"
...
10.211.55.43="1435323298,1800,001c4270603b,ffae8bd02f00020000ab11873bec156a374b12"
```

**Comment by legal90** [comment]
@Tomdarkness thanks for an information! It looks very similar to https://github.com/mitchellh/packer/issues/1746, e.q. it has the same reason. I will try to fix it ASAP.
By the way, what box and guest OS are you using?

**Comment by Tomdarkness** [comment]
@legal90 I'm using AntonioMeireles/coreos-beta which is a CoreOS guest (https://coreos.com/). One property of CoreOS is it automatically updates to the latest version and reboots the VM. I suspect the DUID identifier mentioned in that issue is getting regenerated when it updates, although I'm not sure.

**Comment by legal90** [comment]
Fixed in [v1.4.0](https://github.com/Parallels/vagrant-parallels/releases/tag/v1.4.0)

**Comment by dx9** [comment]
@legal90 Thanks!
